lib/src/transformer/impl_generation.dart

```diff
@@ -167,6 +167,22 @@ class ImplGenerator {
       // ----------------------------------------------------------------------
       generateAccessors(AccessorType.props, declarations.props);
 
+      // Remove $ sign prefixed props mixins when found in the props class with clause.
+      if (declarations.props.node.withClause != null &&
+          declarations.props.node.withClause.mixinTypes.any((type) => type.toString().startsWith('\$'))) {
```

Comment:
I don't think this is going to be sufficient. This entire section is within a `if (declarations.declaresComponent) {` block, and it's only checking with clauses from the props and state classes. But PropsMixins and StateMixins can be used anywhere. I think we'll need to take a slightly different approach where we search every file for with clauses, unfortunately, something like:

1. Check file for class declarations
2. Check mixinTypes for the dual-setup (`<SomeMixinType>` AND `$<SomeMixinType>`)
3. For each pair found, remove the `$`-prefixed mixin

Reply:
Also I think we should be able to add a test case for this (class in a file without a component declaration) that should fail right now